sdk/containerregistry/container-registry/src/blob/models.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { OperationOptions } from "@azure/core-client";
+
+/**
+ * Options for configuring the upload manifest operation.
+ */
+export interface UploadManifestOptions extends OperationOptions {
+  /**
+   * Tag to give the uploaded manifest
+   */
+  tag: string;
+}
+
+/**
+ * The result from uploading a blob to the registry.
+ */
+export interface UploadBlobResult {
+  /**
+   * The digest of the uploaded blob.
+   */
+  digest: string;
+}
+
+/**
+ * The result from downloading a manifest from the registry.
+ */
+export interface DownloadManifestResult {
+  /**
+   * The digest of the downloaded manifest as calculated by the registry.
+   */
+  digest: string;
+
+  /**
+   * The OCI manifest that was downloaded.
+   */
+  manifest: OciManifest;
+
+  /**
+   * The manifest stream that was downloaded.
+   */
+  manifestStream: NodeJS.ReadableStream;
+}
+
+/**
+ * The result from downloading a blob from the registry.
+ */
+export interface DownloadBlobResult {
+  /**
+   * The blob's digest, calculated by the resgistry.
+   */
+  digest: string;
+
+  /**
+   * The blob content.
+   */
+  content: NodeJS.ReadableStream;
+}
+
+/**
+ * The result from uploading a manifest to the registry.
+ */
+export interface UploadManifestResult {
+  /**
+   * The digest of the uploaded manifest, calculated by the registry.
+   */
+  digest: string;
+}
+
+/** Docker V2 image layer descriptor including config and layers. */
+export interface OciBlobDescriptor {
```

Comment:
The optional-ness probably comes from the swagger, but I am wondering whether any of these are required

Reply:
Yup, good point: according to [the spec](https://github.com/opencontainers/image-spec/blob/main/descriptor.md#properties), `mediaType`, `digest` and `size` are required. I'll fix that.